sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/_tests_/e2e/student-scholastic-standings.institutions.controller.saveScholasticStanding.e2e-spec.ts

```diff
@@ -341,26 +449,364 @@ describe("StudentScholasticStandingsInstitutionsController(e2e)-saveScholasticSt
+      },
+    );
+    // Institution token.
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    mockFormioDryRun({
+      studentScholasticStandingChangeType:
+        StudentScholasticStandingChangeType.StudentCompletedProgramEarly,
+      dateOfCompletion: application.currentAssessment.offering.studyStartDate,
+    });
+
+    const endpoint = `/institutions/scholastic-standing/location/${collegeFLocation.id}/application/${application.id}`;
+
+    // Act/Assert
+    let createdScholasticStandingId: number;
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .expect((response) => {
+        expect(response.body.id).toBeGreaterThan(0);
+        createdScholasticStandingId = +response.body.id;
+      });
+
+    const studentScholasticStanding = await db.studentScholasticStanding.exists(
+      {
+        select: {
+          id: true,
+        },
+        relations: {
+          application: true,
+        },
+        where: {
+          id: createdScholasticStandingId,
+          application: { id: application.id },
+        },
+      },
+    );
+
+    expect(studentScholasticStanding).toBe(true);
+
+    const notifications = await db.notification.find({
+      select: {
+        id: true,
+        user: { id: true },
+        notificationMessage: { id: true },
+      },
+      relations: { user: true, notificationMessage: true },
+      where: { user: { id: application.student.user.id } },
+      order: { notificationMessage: { id: "ASC" } },
+    });
+    expect(notifications).toEqual([
+      {
+        id: expect.any(Number),
+        notificationMessage: {
+          id: NotificationMessageType.InstitutionReportsChange,
+        },
+        user: { id: application.student.user.id },
+      },
+    ]);
+  });
+
+  it("Should not create new scholastic standing 'School transfer' for a part-time application when date of change is not between study start date and end date.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(
+      db.dataSource,
+      {
+        institutionLocation: collegeFLocation,
+      },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // Institution token.
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const endpoint = `/institutions/scholastic-standing/location/${collegeFLocation.id}/application/${application.id}`;
+
+    mockFormioDryRun({
+      validDryRun: false,
+      studentScholasticStandingChangeType:
+        StudentScholasticStandingChangeType.SchoolTransfer,
+      dateOfChange: addToDateOnlyString(
+        application.currentAssessment.offering.studyStartDate,
+        1,
+        "day",
+      ),
+    });
```

Comment:
Date of change being 1 day after study start date is absolutely valid. 

Do you mean the study end date?

Reply:
Same for date of withdrawal and date of completion in other tests.